src/components/posts/PostList.js

```diff
@@ -0,0 +1,36 @@
+import React, { useContext, useEffect } from "react"
+import { PostContext } from "./PostProvider"
+import { Link, useParams } from "react-router-dom"
+import "./Post.css"
+
+export const PostList = () => {
+    const { posts, getPosts, getPostsByUserId } = useContext(PostContext)
+
+    const { user_id } = useParams()
+
+    useEffect(() => {
+        if (user_id > 0) {
+            getPostsByUserId(user_id)
+        } else {
+            getPosts()
+        }
+    }, [])
+
+    useEffect(() => {
+        console.log(posts)
+    }, [posts])
+
+    return (
+        <>
+            <div>
+                {posts.map(post =>
+                    <div className="post_card">
```

Comment:
Warning suggest adding a key.

Reply:
Just pushed up this change. Thanks for catching this!